### PR TITLE
csi: enhance error messages

### PR DESCRIPTION
**Pull request by yati1998**
this commit enhances the error message to
give detail info to the users.

### CODE REVIEW

File: pkg/filesystem/subvolume.go [hunk 1]
```diff
@@ -80,13 +80,13 @@ func getExternalClusterDetails(ctx context.Context, clientsets *k8sutil.Clientse
...
 	if err != nil {
-		logging.Fatal(fmt.Errorf("Error fetching secrets: %v\n", err))
+		logging.Fatal(fmt.Errorf("Error fetching secrets in namespace %q: %v", clusterNamespace, err))
 	}
 	for i := range scList.Items {
 		if strings.HasPrefix(scList.Items[i].ObjectMeta.Name, "rook-csi-cephfs-provisioner") {
 			data := scList.Items[i].Data
 			if data == nil {
-				logging.Fatal(fmt.Errorf("Secret data is empty"))
+				logging.Fatal(fmt.Errorf("Secret data is empty for %q/rook-csi-cephfs-provisioner", clusterNamespace))
```
[the remark this answers is not in the document]
Comment (subhamkrai) [reply pkg/filesystem/subvolume.go:89]: @yati1998 please check this, make sense to me aswell

**Review by subhamkrai — APPROVED** [review]

File: pkg/filesystem/subvolume.go [hunk 2]
```diff
@@ -80,13 +80,13 @@ func getExternalClusterDetails(ctx context.Context, clientsets *k8sutil.Clientse
...
 	if err != nil {
-		logging.Fatal(fmt.Errorf("Error fetching secrets: %v\n", err))
+		logging.Fatal(fmt.Errorf("Error fetching secrets in namespace %q: %v", clusterNamespace, err))
 	}
 	for i := range scList.Items {
 		if strings.HasPrefix(scList.Items[i].ObjectMeta.Name, "rook-csi-cephfs-provisioner") {
 			data := scList.Items[i].Data
 			if data == nil {
-				logging.Fatal(fmt.Errorf("Secret data is empty"))
+				logging.Fatal(fmt.Errorf("Secret data is empty for %v/rook-csi-cephfs-provisioner", clusterNamespace))
```
Comment (travisn) [inline pkg/filesystem/subvolume.go:89]: `%v` is used when printing a struct or other complex type. If it's just a string, please use `%s` when no quotes are needed. (same for other changes in this PR)
```suggestion
				logging.Fatal(fmt.Errorf("Secret data is empty for %s/rook-csi-cephfs-provisioner", clusterNamespace))
```

**Review by travisn — CHANGES_REQUESTED** [review]

**Review by travisn — APPROVED** [review]

**Review by subhamkrai — CHANGES_REQUESTED** [review]
@yati1998 please check the CI issues. Thanks

File: pkg/filesystem/subvolume.go [hunk 3]
```diff
@@ -80,13 +80,13 @@ func getExternalClusterDetails(ctx context.Context, clientsets *k8sutil.Clientse
...
 	if err != nil {
-		logging.Fatal(fmt.Errorf("Error fetching secrets: %v\n", err))
+		logging.Fatal(fmt.Errorf("Error fetching secrets in namespace %q: %v", clusterNamespace, err))
 	}
 	for i := range scList.Items {
 		if strings.HasPrefix(scList.Items[i].ObjectMeta.Name, "rook-csi-cephfs-provisioner") {
 			data := scList.Items[i].Data
 			if data == nil {
-				logging.Fatal(fmt.Errorf("Secret data is empty"))
+				logging.Fatal(fmt.Errorf("Secret data is empty for %s/rook-csi-cephfs-provisioner", clusterNamespace))
```
Comment (Madhu-1) [inline pkg/filesystem/subvolume.go:89]:
```suggestion
logging.Fatal(fmt.Errorf("Secret data is empty for %s/%s", clusterNamespace,scList.Items[i].ObjectMeta.Name))
```

**Review by Madhu-1 — APPROVED** [review]

**Review by subhamkrai — APPROVED** [review]

**Pull request merged**